src/libraries/System.Private.CoreLib/src/System/Single.cs

```diff
@@ -1469,13 +1469,8 @@ private static bool TryConvertTo<TOther>(float value, [MaybeNullWhen(false)] out
             else if (typeof(TOther) == typeof(nuint))
             {
 #if MONO
-#if TARGET_64BIT
-                nuint actualResult = (value >= ulong.MaxValue) ? unchecked((nuint)ulong.MaxValue) :
-                                     (value <= ulong.MinValue) ? unchecked((nuint)ulong.MinValue) : (nuint)value;
-#else
-                nuint actualResult = (value >= uint.MaxValue) ? uint.MaxValue :
-                                     (value <= uint.MinValue) ? uint.MinValue : (nuint)value;
-#endif
+                nuint actualResult = (value >= uint.MaxValue) ? nuint.MaxValue :
```

Comment:
- This has a typo (it needs to use nuint everywhere)
- This ifdef may be intentional. Unlike ulong.Maxnuint.MaxValue / nuint.MinValue are properties, so this may be a regression on Mono interpreter.

Reply:
I would leave this as is.

Reply:
> This ifdef may be intentional. Unlike ulong.MaxValue and friends, nuint.MaxValue / nuint.MinValue are properties, so this may be a regression on Mono interpreter.

It would be good to know if this isn't being handled on Mono Interpreter today. It is likely a small change to recognize it and would avoid any issues in the core libraries or user code

Reply:
The ifdef here was notably because Mono didn't have the correct conversion behavior at the time.

I believe that has since been rectified.